### ISSUE TITLE
Shortest path functions

### ISSUE THREAD

**Issue by tpoisot** [PoisotLab/EcologicalNetworks.jl#104]
Additional path algorithms (in progress in `feature-path-algorithms`)

- [x] Bellman-Ford
- [ ] Improve the speed for Bellman-Ford by calling `interactions` only once
- [ ] Djikstra

**Comment by MichielStock** [comment]
I have implemented Dijkstra for my optimization course. If you want I can implement this for this package?

**Comment by MichielStock** [comment]
Dijkstra can be implemented implemented easily with a [time complexity](https://en.wikipedia.org/wiki/Dijkstra%27s_algorithm#Running_time) of O(|V|^2) without additional data structures. If the O(|E|+|V|\log |V|) version is needed, I will need a heap from DataStructures.jl.

Do you think that improving the time complexity justifies adding another dependency? Take note that all networks are likely going to be small.

**Comment by MichielStock** [comment]
At this point, I implemented Dijkstra from source to all nodes. It is substantially faster than Bellman-Ford for larger (> 100 nodes) networks. 

I also implemented a version of Dijkstra which computes the distance of all nodes to all other nodes in a single pass to the graph. However, this does not seem to be what you are doing or want to attain?

However, I did managed to reproduce the result of BF using Dijkstra. It is quite a bit faster than the original implementation (at least from modest network sizes onwards), but I still think it is relatively slow. 

See [source code](https://github.com/MichielStock/EcologicalNetworks.jl/blob/dijkstra/src/community/paths.jl).
